object_detection/ObjectDetection.py

```diff
@@ -0,0 +1,149 @@
+        # print("Yolo IDs:", detections)
+
+        for cls, yolo_id in detections:
+            current_yolo_ids[cls].add(yolo_id)
+
+            if yolo_id in self.tracker_ids[cls]:
+                tracker_id = self.tracker_ids[cls][yolo_id]
+                self.disappeared[cls][tracker_id] = 0
+            else:
+                if yolo_id != -1:
+                    new_tracker_id = self.get_next_available_id(cls)
+                    self.tracker_ids[cls][yolo_id] = new_tracker_id
+                    self.disappeared[cls][new_tracker_id] = 0
+                else:
+                    pass
+
+        for cls in self.classifications:
+            for yolo_id, tracker_id in list(self.tracker_ids[cls].items()):
+                if yolo_id not in current_yolo_ids[cls]:
+                    self.disappeared[cls][tracker_id] += 1
+                    if self.disappeared[cls][tracker_id] > self.max_disappeared:
+                        del self.tracker_ids[cls][yolo_id]
+                        del self.disappeared[cls][tracker_id]
+
+        return self.tracker_ids
+
+    def get_next_available_id(self, cls):
+        used_ids = set(self.tracker_ids[cls].values())
+        tracker_id = 1
+        while tracker_id in used_ids:
+            tracker_id += 1
+        return tracker_id
+
+    def get_tracker_id(self, cls, yolo_id):
+        return self.tracker_ids[cls].get(yolo_id, None)
+
+def convertB64ToFrame(b64_string):
+    jpg_original = base64.b64decode(b64_string)
+    jpg_as_np = np.frombuffer(jpg_original, dtype=np.uint8)
+    img = cv2.imdecode(jpg_as_np, cv2.IMREAD_COLOR)
+    return img
+
+def convertFrameToB64(frame):
+    _, buffer = cv2.imencode('.jpg', frame)
+    return base64.b64encode(buffer).decode()
+
+def get_model_and_tracker(camera_id, class_assignments):
+    if camera_id not in models:
+        models[camera_id] = YOLO("assets/yolo11s.onnx", task='detect')
+        id_trackers[camera_id] = IDTracker(class_assignments.keys())
+    return models[camera_id], id_trackers[camera_id]
+
+def read_class_names(class_file_name): #loads class names from file
+    names = {}
+    with open(class_file_name, "r") as data:
+        for ID, name in enumerate(data):
+            names[ID] = name.strip("\n")
+    return names
+
+def get_class_assignments(objects): #returns a dictionary of class assignments
+    names = read_class_names("assets/coco.names")
+    class_assignment = {}
+    for key, value in names.items():
+        if value in objects:
+            class_assignment[value] = key  #Use string label as key, numeric ID as value
+    return class_assignment
+
+def detect_objects(input_frame, camera_id, task_settings):
+    objects = task_settings.get("objects_to_detect")
+    class_assignments = get_class_assignments(objects)
+    
+    model, id_tracker = get_model_and_tracker(camera_id, class_assignments)
+    
+    enable_tracking = task_settings.get("object_tracking", True)
```

Comment:
lets change the default to False and task_settings will also have `enable_tracking` in the settings.